### ISSUE TITLE
[WhatsApp] all chats disappeared

### ISSUE THREAD

**Issue by tibtab** [loqui/im#746]
I had really strange problem this morning. Loqui was running in background, and when I reenterd it all WhatsApp chats and groupchats were disappeared:
![2015-06-13-11-20-31](https://cloud.githubusercontent.com/assets/11966750/8144269/f78806ce-11d5-11e5-9f8d-4daa0ae61fe3.png)
I restarted loqui, restarted the phone, nothing helped. Nevertheless I received new messages:
![2015-06-13-11-23-58](https://cloud.githubusercontent.com/assets/11966750/8144273/5b10751e-11d6-11e5-8048-9b7d8688b890.png)
I tried to start a new chat; the phone switched to the screen displayed in the first picture, but through swiping to the left I could enter the chat, and old messages were displayed but although it said that I now only have one unread notification left, the new message wasn't shown in the chat.
I finally deleted the account and recreated it and now it works, but I thought I should tell you this issue.

**Comment by tibtab** [comment]
I had this problem 3 times by now. Maybe it only happens when I open or start loqui, first my xmpp account is displayed and than I change to whatsapp using the account preview also shown in the picture above, but I'm not sure.

**Comment by TitanNano** [comment]
this issue appears when you use the account switcher in the header?

**Comment by tibtab** [comment]
Yes i think so but I'm not sure if that was the problem or something else

**Comment by TitanNano** [comment]
can you try to figure out how to reproduce it?

**Comment by tibtab** [comment]
Add two accounts to loqui first the xmpp that it is the first account wich is displayed when you start.
Close loqui, let someone send you an whatsapp message and restart it.
No swich the account through the header.
Im not sure if it works like this because i dont want to provoke that issue while writing with people which may get tired of sending there messages me twice or more because i didn't receive them :/

**Comment by TitanNano** [comment]
well this also happens sometimes to me, but not very often... You are
experiencing this recently? You can try to reconnect, most of the time you
will get the message then.

2015-06-16 17:14 GMT+02:00 tibtab notifications@github.com:

**Comment by ghost** [comment]
@tibtab If possible and available get a new phone number just for testing, e.g. a free prepaid SIM card (i also can give you access to some of my testing phone numbers). And try to reproduce the issue with Firefox OS Simulator.

**Comment by tibtab** [comment]
I don't know any provider who gives his phone numbers away for free.

**Comment by tibtab** [comment]
Have this problem again right now, I changed the account through the sidebar menu, so this disprooved my supposition.
I opened the webIDE should I look for something special?

**Comment by TitanNano** [comment]
uh I don't really know. you can just post the log here, I will take a look
at it.

2015-06-19 20:34 GMT+02:00 tibtab notifications@github.com:

**Comment by tibtab** [comment]
how to get that log?

**Comment by TitanNano** [comment]
uhm just copy it from the console...

2015-06-19 20:40 GMT+02:00 tibtab notifications@github.com:

**Comment by tibtab** [comment]
This ?

```
"[l10n] #contactAdd is undefined." l10n.js:68
"SHOW" "main" undefined undefined menu.js:297
"Write node called with  {"_tag":"presence","children":[],"attributes":{"type":"unavailable"},"_data":null} ~ self._writeNode@app://f1c3f984-4611-48ed-a81d-64cc5877694e/scripts/mozillahispano/coseme.js:7569 [ConnectionManager]" coseme.js:3541
"Write node called with  {"_tag":"iq","children":[{"_tag":"status","children":[],"attributes":{},"_data":"Benutze Loqui IM auf FirefoxOS"}],"attributes":{"to":"s.whatsapp.net","type":"set","id":"sendsetstatus_1","xmlns":"status"},"_data":null} ~ self._writeNode@app://f1c3f984-4611-48ed-a81d-64cc5877694e/scripts/mozillahispano/coseme.js:7569 [ConnectionManager]" coseme.js:3541
"Write node called with  {"_tag":"presence","children":[],"attributes":{"type":"available"},"_data":null} ~ self._writeNode@app://f1c3f984-4611-48ed-a81d-64cc5877694e/scripts/mozillahispano/coseme.js:7569 [ConnectionManager]" coseme.js:3541
"Write node called with  {"_tag":"iq","children":[{"_tag":"status","children":[],"attributes":{},"_data":"Benutze Loqui IM auf FirefoxOS"}],"attributes":{"to":"s.whatsapp.net","type":"set","id":"sendsetstatus_2","xmlns":"status"},"_data":null} ~ self._writeNode@app://f1c3f984-4611-48ed-a81d-64cc5877694e/scripts/mozillahispano/coseme.js:7569 [ConnectionManager]" coseme.js:3541
[object Object] account.js:283
[object Object] account.js:283
"Write node called with  {"_tag":"presence","children":[],"attributes":{"type":"unavailable"},"_data":null} ~ self._writeNode@app://f1c3f984-4611-48ed-a81d-64cc5877694e/scripts/mozillahispano/coseme.js:7569 [ConnectionManager]" coseme.js:3541
"Write node called with  {"_tag":"iq","children":[{"_tag":"status","children":[],"attributes":{},"_data":"Benutze Loqui IM auf FirefoxOS"}],"attributes":{"to":"s.whatsapp.net","type":"set","id":"sendsetstatus_3","xmlns":"status"},"_data":null} ~ self._writeNode@app://f1c3f984-4611-48ed-a81d-64cc5877694e/scripts/mozillahispano/coseme.js:7569 [ConnectionManager]" coseme.js:3541
"Write node called with  {"_tag":"presence","children":[],"attributes":{"type":"available"},"_data":null} ~ self._writeNode@app://f1c3f984-4611-48ed-a81d-64cc5877694e/scripts/mozillahispano/coseme.js:7569 [ConnectionManager]" coseme.js:3541
"Write node called with  {"_tag":"iq","children":[{"_tag":"status","children":[],"attributes":{},"_data":"Benutze Loqui IM auf FirefoxOS"}],"attributes":{"to":"s.whatsapp.net","type":"set","id":"sendsetstatus_4","xmlns":"status"},"_data":null} ~ self._writeNode@app://f1c3f984-4611-48ed-a81d-64cc5877694e/scripts/mozillahispano/coseme.js:7569 [ConnectionManager]" coseme.js:3541
"Write node called with  {"_tag":"presence","children":[],"attributes":{"type":"unavailable"},"_data":null} ~ self._writeNode@app://f1c3f984-4611-48ed-a81d-64cc5877694e/scripts/mozillahispano/coseme.js:7569 [ConnectionManager]" coseme.js:3541
"Write node called with  {"_tag":"iq","children":[{"_tag":"status","children":[],"attributes":{},"_data":"Benutze Loqui IM auf FirefoxOS"}],"attributes":{"to":"s.whatsapp.net","type":"set","id":"sendsetstatus_5","xmlns":"status"},"_data":null} ~ self._writeNode@app://f1c3f984-4611-48ed-a81d-64cc5877694e/scripts/mozillahispano/coseme.js:7569 [ConnectionManager]" coseme.js:3541
```

**Comment by TitanNano** [comment]
yes. is that all you have?

**Comment by tibtab** [comment]
Yes, but I not wasn't connected with the webIDE until I had the problem.

After a restart of the programm only:

"[l10n] #contactAdd is undefined." l10n.js:68
"SHOW" "main" undefined undefined menu.js:297

is displayed, not more.

**Comment by TitanNano** [comment]
yes that is fine. :-/

2015-06-19 20:52 GMT+02:00 tibtab notifications@github.com:

**Comment by tibtab** [comment]
Could it be caused by any changes you did in "fixes #706" ?
Because I never ha this Problem before.

**Comment by TitanNano** [comment]
okay I fixed this. you should have received a message without any sender jid... I don't know how this happened, but the same just happened to me too.
